src/lib/nftClaimer/utils.ts

```diff
@@ -132,3 +137,43 @@ export function validateAddresses(addresses: Record<string, string>) {
 
   return true;
 }
+
+export async function snapshotFee() {
+  const storage = storageEngine('nft-claimer');
+  const keyname = 'snapshotFee';
+  const CACHE_TTL = 60 * 60 * 1000; // 1 hour
+
+  try {
+    const cache = await storage.get(keyname);
+    if (typeof cache === 'string') {
+      const result = JSON.parse(cache);
+
+      if (result.lastUpdate > +new Date() - CACHE_TTL) {
+        return result.snapshotFee;
+      }
+    }
+
+    const providerOptions: Record<string, string> = {};
+    ['etherscan'].forEach(key => {
+      const envValue = process.env[`${key.toUpperCase()}_API_KEY`];
+      if (!!envValue) {
+        providerOptions[key] = envValue;
+      }
+    });
+
+    const provider = getDefaultProvider(NFT_CLAIMER_NETWORK, providerOptions);
```

Comment:
We should use brovider here instead right?

Reply:
Was not aware brovider can do that, will take a look